### PR TITLE
Restructure endpoint generator

### PR DESCRIPTION
**Pull request by shivahari**
I have made the following changes to the `API Endpoint Generator` project:

1. I replaced the `utils/generate_endpoint_from_openapi_spec.py` file with the following modules:
- `api_auto_generator/endpoint_module_generator.py`
- `api_auto_generator/endpoint_name_generator.py`
- `api_auto_generator/openapi_spec_parser.py`

2. I have added a `api_auto_generator/templates` directory to house the `Jinja2` template file

3. I have added a `api_auto_generator/Endpoint_Generator.md` file to detail instructions on:
- How to run the script
- How the script works
- Limitations & Constraints

### CODE REVIEW

File: api_auto_generator/templates/endpoint_template.jinja2 [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                :{{data_param[0]}}: {{data_param[1]}}
+        {%- endif %}
+        {%- endfor %}
+        {%- endif %}
+         """
+        url = self.{{class_content['url_method_name']}}(f"{{function_value['endpoint']}}")
+        json_response = self.make_request(method='{{function_value["http_method"]}}', url=url{{function_value['base_api_param_string']}})
+        return {
+            'url' : url,
+            'response' : json_response['json_response']
```
Comment (akkuldn) [inline api_auto_generator/templates/endpoint_template.jinja2:70]: Is the key name going to be ['json_response' ] for all endpoints?

**Review by akkuldn — CHANGES_REQUESTED** [review]
Hey @shivahari ,  i have added the review comments. Also please fix the Codacy issues.

File: api_auto_generator/openapi_spec_parser.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        }
+        """
+        requestbody_type = None
+        requestbody_param = []
+        content_schema_type = None
+
+        # Parse the request_body in the spec
+        # Identify the content type of the request_body
+        # This module currently supports: 1.json, 2.form content types
+        for content in request_body.content:
```
[the remark this answers is not in the document]
Comment (shivahari) [reply api_auto_generator/openapi_spec_parser.py:153]: I contemplated `nested-ifs` here. I even tried breaking it to smaller methods, but:
- it increased complexity
- I felt making future additions will get tougher.

File: api_auto_generator/openapi_spec_parser.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+
+    # pylint: disable=too-few-public-methods
+    def __init__(self, spec_file: TextIO, logger_obj) -> None:
+        "Init Spec Parser Obj"
+
+        self.logger = logger_obj
+        # Generate Final dict usable against a Jinja2 template from the OpenAPI Spec
+        self._fdict = {}
+        # _fdict structure:
```
[the remark this answers is not in the document]
Comment (shivahari) [reply api_auto_generator/openapi_spec_parser.py]: I have replaced the comments `#` in multiple line with multi-line comment `""""""`

File: api_auto_generator/templates/endpoint_template.jinja2 (hunk 1 above)
Comment (shivahari) [reply api_auto_generator/templates/endpoint_template.jinja2:70]: Yes. `json_reponse` is they key we use already in our `Endpoint` file.
This needs to change, but in a different PR targeting all `Endpoint` files.

**Pull request closed without merging**